### PR TITLE
Social networking features

### PR DESCRIPTION
**Pull request by mongkuen**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class UsersController < ApplicationController
-  before_action :redirect_to_home
+  before_action :redirect_to_home, only: [:new, :create]
```
Comment (brandoncc) [inline app/controllers/users_controller.rb]: This method name could be improved. The way it reads, it redirects every user to /home if they try to go to users/new or post to users....I doubt that is really what happens though. I would think about renaming that to describe what it checks for.

File: app/models/user.rb [hunk 2]
```diff
@@ -22,4 +26,17 @@ def normalize_queue_item_positions
       item.update_attributes(position: index + 1)
     end
   end
+
+  def followable?(leader)
+    self.not_yet_followed?(leader) && self.is_not_leader?(leader)
```
Comment (brandoncc) [inline app/models/user.rb]: `self` is not needed in these three methods.

File: app/models/user.rb [hunk 3]
```diff
@@ -22,4 +26,17 @@ def normalize_queue_item_positions
...
+  end
+
+  def not_yet_followed?(leader)
+    !self.leaders.include?(leader)
+  end
+
+  def is_not_leader?(leader)
```
Comment (brandoncc) [inline app/models/user.rb:38]: I don't think this method is needed. You are abstracting `self != leader`. I don't think that warrants abstraction and the indirection that comes with it.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (mongkuen) [inline app/controllers/users_controller.rb]: I will change the method name!

File: app/models/user.rb (hunk 3 above)
Comment (mongkuen) [inline app/models/user.rb:38]: okay. I'll just shove that logic into #followable? then.

**Pull request merged**